net/net.go

```diff
@@ -234,12 +234,37 @@ func (n *net) CreateThread(_ context.Context, id thread.ID, opts ...core.NewThre
 	return n.getThreadWithAddrs(id)
 }
 
-func (n *net) ensureUnique(id thread.ID) error {
-	_, err := n.store.GetThread(id)
-	if err == nil {
-		return lstore.ErrThreadExists
+func (n *net) ensureUnique(id thread.ID, key crypto.Key) error {
```

Comment:
This is now a lot more verbose, as there are multiple different possible scenarios here, but I think the early outs help.